src/core/main.c

```diff
@@ -694,22 +694,26 @@ static void set_manager_defaults(Manager *m) {
         (void) manager_transient_environment_add(m, arg_default_environment);
 }
 
-static void set_manager_settings(Manager *m) {
+static void set_manager_settings(Manager *m, bool reload) {
 
         assert(m);
 
-        /* Propagates the various manager settings into the manager object, i.e. properties that effect the manager
-         * itself (as opposed to just being inherited into newly allocated units, see set_manager_defaults() above). */
+        /* Propagates the various manager settings into the manager object, i.e. properties that
+         * effect the manager itself (as opposed to just being inherited into newly allocated
+         * units, see set_manager_defaults() above). */
 
-        m->confirm_spawn = arg_confirm_spawn;
-        m->service_watchdogs = arg_service_watchdogs;
-        m->runtime_watchdog = arg_runtime_watchdog;
-        m->reboot_watchdog = arg_reboot_watchdog;
-        m->kexec_watchdog = arg_kexec_watchdog;
-        m->cad_burst_action = arg_cad_burst_action;
+        if (!reload) {
+                m->confirm_spawn = arg_confirm_spawn;
+                m->service_watchdogs = arg_service_watchdogs;
+                m->reboot_watchdog = arg_reboot_watchdog;
+                m->kexec_watchdog = arg_kexec_watchdog;
+                m->cad_burst_action = arg_cad_burst_action;
 
-        manager_set_show_status(m, arg_show_status, "commandline");
-        m->status_unit_format = arg_status_unit_format;
+                manager_set_show_status(m, arg_show_status, "commandline");
+                m->status_unit_format = arg_status_unit_format;
+        }
```

Comment:
what's the reason for avoiding these on reload?

if there's a reason to avoid them, this should probably be encapsulated in separate calls instead of passing in the bool

i.e. have both set_manager_settings_once() and set_manager_settings_always() or so, where the former is called only once per exec cycle, but the latter once each for each reload cycle. or something like this

Reply:
No reason, I think all of them can be updated on reload. I'll change it.

Reply:
Just out of curiosity, this question is not really needed since we don't need to pass a bool anymore but why would you have preferred 2 separate calls over passing in the bool ?